src/core/common/message.cpp

```diff
@@ -143,30 +149,18 @@ void MessagePool::FreeBuffers(Buffer *aBuffer)
     }
 }
 
-otError MessagePool::ReclaimBuffers(int aNumBuffers, Message::Priority aPriority)
+otError MessagePool::ReclaimBuffers(Message::Priority aPriority)
 {
-#if OPENTHREAD_MTD || OPENTHREAD_FTD
-    while (aNumBuffers > GetFreeBufferCount())
-    {
-        SuccessOrExit(Get<MeshForwarder>().EvictMessage(aPriority));
-    }
-
-exit:
-#else
-    OT_UNUSED_VARIABLE(aPriority);
-#endif
-
-    // First comparison is to get around issues with comparing
-    // signed and unsigned numbers, if aNumBuffers is negative then
-    // the second comparison wont be attempted.
-    return (aNumBuffers < 0 || aNumBuffers <= GetFreeBufferCount()) ? OT_ERROR_NONE : OT_ERROR_NO_BUFS;
+    return Get<MeshForwarder>().EvictMessage(aPriority);
```

Comment:
The function `EvictMessage` only removes a message. The space left after releasing a message may not be enough for the new message. Especially when the traffic is heavy, it may cause insufficient space to create a MLE message. Which may cause the link to be broken

Reply:
it's wrapped in a `ReclaimBuffer` method and the method is used in a while loop until enough buffer is got.